spec/services/atmosphere/cloud/satisfy_appliance_spec.rb

```diff
@@ -9,16 +9,17 @@
     allow(Atmosphere::Optimizer.instance).to receive(:run)
   end
 
-  let!(:wf) { create(:workflow_appliance_set) }
-  let!(:wf2) { create(:workflow_appliance_set) }
-  let!(:shareable_appl_type) { create(:shareable_appliance_type) }
   let!(:fund) { create(:fund) }
+  let!(:u) { create(:user, funds: [fund]) }
+  let!(:wf) { create(:workflow_appliance_set, user: u) }
+  let!(:wf2) { create(:workflow_appliance_set, user: u) }
+  let!(:shareable_appl_type) { create(:shareable_appliance_type) }
```

Comment:
What four objects? I only see one extra object (`u`). Actually, this change results in **fewer** objects being created due to the way factories work (e.g. if you don't specify `user: u` in `ApplianceSet` creators then FactoryGirl will happily create an entirely new `User` object for each `ApplianceSet` etc.

Oh, and by the way - skipping `u` causes most of the tests in this spec to fail because the "default" users which FG will create are not bound to the correct funds (and therefore fail the new optimizer checks).